python/ray/worker.py

```diff
@@ -1658,7 +1669,9 @@ def _init(address_info=None,
             plasma_directory=plasma_directory,
             huge_pages=huge_pages,
             include_webui=include_webui,
-            use_raylet=use_raylet)
+            use_raylet=use_raylet,
+            plasma_store_socket_name=plasma_store_socket_name,
+            temp_dir=tempfile_services.get_temp_root())
```

Comment:
in the `else` below, check that `temp_dir` is `None` and raise an exception if not

Reply:
same with `plasma_store_socket_name`